http_server.py

```diff
@@ -25,29 +56,33 @@ def parse_request(request):
                 return uri
             else:
                 raise SyntaxError("400 Bad Request")
-        else:
+        elif method in ["POST", "UPDATE", "DELETE", "PUT",
+                        "HEAD", "OPTIONS", "CONNECT"]:
             raise ValueError("405 Method Not Allowed")
+        else:
+            raise SyntaxError("400 Bad Request")
```

Comment:
nice parsing of the difference between a valid but unsupported HTTP method and an invalid method.